.pre-commit-config.yaml

```diff
@@ -23,10 +23,24 @@ repos:
     rev: v1.5.1
     hooks:
     -   id: python-use-type-annotations
-# -   repo: https://github.com/pre-commit/mirrors-mypy
-#     rev: v0.782
-#     hooks:
-#     -   id: mypy
+-   repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v0.782
+    hooks:
+    -   id: mypy
+        exclude: src/syft/proto
+        args: [
+            "--ignore-missing-imports",
+            "--scripts-are-modules",
+            "--disallow-incomplete-defs",
+            "--no-implicit-optional",
+            "--warn-unused-ignores",
+            "--warn-redundant-casts",
+            "--strict-equality",
+            # "--warn-unreachable",
```

Comment:
Q: we don't want those?

Reply:
Are you asking why there are commended out ones?
The ones that are commented out are strict enough to cause a lot more issues which I can fix but its better to do these things in batches.

I left them there so they are easy to toggle during development, and as a reminder that we should probably finish them as well.

I only saw 1 error for --warn-unreachable but it could be that the code is just incomplete.
The untyped decorators / calls will mostly be covered by --disallow-untyped-defs but that means a lot of:
```
- def function(self):
+ def function(self) -> None:
```

I was able to get all the other flags complete in time for this PR to be merged at the start of the work day for everyone on EU time zones which is preferable than having a big noisy PR that stays open across other merges.